apps/console/src/features/connections/api/authenticators.ts

```diff
@@ -0,0 +1,385 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { AsgardeoSPAClient, HttpClientInstance } from "@asgardeo/auth-react";
+import useResourceEndpoints from "@wso2is/common/src/hooks/use-resource-endpoints";
+import { IdentityAppsApiException } from "@wso2is/core/exceptions";
+import { HttpMethods } from "@wso2is/core/models";
+import { AxiosError, AxiosRequestConfig, AxiosResponse } from "axios";
+import { store } from "../../core";
+import useRequest, {
+    RequestConfigInterface,
+    RequestErrorInterface,
+    RequestResultInterface
+} from "../../core/hooks/use-request";
+import { ConnectionManagementConstants } from "../constants/connection-constants";
+import { 
+    AuthenticatorInterface, 
+    MultiFactorAuthenticatorInterface 
+} from "../models/authenticators";
+import { 
+    ConnectionInterface, 
+    FederatedAuthenticatorListItemInterface, 
+    FederatedAuthenticatorListResponseInterface, 
+    FederatedAuthenticatorMetaInterface 
+} from "../models/connection";
+
+/**
+ * Get an axios instance.
+ */
+const httpClient: HttpClientInstance = AsgardeoSPAClient.getInstance()
+    .httpRequest.bind(AsgardeoSPAClient.getInstance())
+    .bind(AsgardeoSPAClient.getInstance());
```

Comment:
Binding twice is not necessary

Reply:
As these changes are directly brought from previous PRs, will continue as it is.